### PR TITLE
fix: Remove multiple Script syntax

### PR DESCRIPTION
**Pull request by EmosewaMC**
squashed and rebased from #1489 

The client physically does not support having multiple scripts on a GameObject. This PR seeks to remove implying the possibility of this being a thing.

tested:
entering, playing and leaving avant gardens survival (single player, played start to finish)
Using pirate hook build (OnStartup)
red blocks smashables cycle (OnDie, OnQuickbuildComplete)
spider queen instance battle functions from start to finish (OnStartup, ondie, otimer done, OnHitOrHealResult, OnUpdate)
entering fv for the first time (OnPhantomCollision)
building fv canister for blue brick (notifyObject)
boogie down functions (onemote received)
property tutorial missions (OnZonePropertyModelRotated, OnZonePropertyModelRemovedWhileEquipped, OnZonePropertyModelRemoved, OnZonePropertyModelPickedUp, OnZonePropertyModelPlaced, OnZonePropertyModelEquipped, OnZonePropertyEditBegin, OnZonePropertyEditEnd)
growing flowers on crux (OnSkillEventFired)
Imaginite pack (OnHit)
damaged by fire (OnPhantomCollision, OnOffPhantomCollision)
building both consoles fv (OnQuickbuildNotifyState)
Mast teleporting (OnQuickBuildComplete)
fans in ag turn off and on (OnFireEventServerSide trigger)
naxue naomi breadcrumbs mission (OnMissionDialogueOK)
nexus jay join faction mission (OnRespondToMission)
build rocket for mardolf the orange (OnModularBuildExit)
Using a ninjago ice rail activator in caves (OnPlayerRailActivated)

### CODE REVIEW

**Review by aronwk-aaron — APPROVED** [review]

**Review by aronwk-aaron — APPROVED** [review]

File: dGame/Entity.cpp [hunk 1]
```diff
@@ -146,17 +146,15 @@ Entity::~Entity() {
 			return;
 		}
 
-		Entity* zoneControl = Game::entityManager->GetZoneControlEntity();
-		for (CppScripts::Script* script : CppScripts::GetEntityScripts(zoneControl)) {
-			script->OnPlayerExit(zoneControl, this);
+		auto* zoneControl = Game::entityManager->GetZoneControlEntity();
```
Comment (jadebenn) [inline dGame/Entity.cpp:149]: Why the change to auto* here? Not an issue, just curious.

Comment (EmosewaMC) [reply dGame/Entity.cpp:149]: idk, just felt like it

**Review by jadebenn — APPROVED** [review]
lgtm
